src/QtHardMon.cpp

```diff
@@ -881,9 +881,10 @@ void QtHardMon::registerClicked(QTreeWidgetItem* registerItem) {
   // or if the register is write-only
 
   PreferencesProvider& preferencesProvider = PreferencesProviderSingleton::Instance();
-  DeviceElementQTreeItem* selectedItem = static_cast<DeviceElementQTreeItem*>(registerItem);
+  auto* selectedItem = dynamic_cast<DeviceElementQTreeItem*>(registerItem);
```

Comment:
There is no point in dynamic_casting and the not doing a nullptr check.
The dynamic cast here cannot fail, so we can as well use a static cast.

Reply:
That was a complaint from the linter

Reply:
maybe add an assert?

Reply:
Added an assert